### PR TITLE
chore(internal): Deprecate V4 config and remove intermediate types

### PR DESCRIPTION
**Pull request by hikchoi**
WIP

# Pull Request Checklist

If you are a community contributor, copy and paste the PR template from [Dendron Community PR Checklist](https://gist.github.com/kevinslin/5d1a6663638259e7dbd33b01975cc00f) and add it to the body of the pull request. 

If you are a team member, copy and paste the template from [Dendron Extended PR Checklist](https://gist.github.com/kevinslin/dfc7a101f21c58478216aa0d70256bc2).

To copy the template, click on the `Raw` button on the gist to copy the plaintext version of the template to this PR.

### CODE REVIEW

**Comment by hikchoi** [comment]
NOTE: this PR still has some pending tasks to be done, but since the diffs are getting huge it would be great to do an initial pass of reviews at this point.

pending tasks:
- [ ] gracefully handle v4 configs without forcing or prompting for a migration

File: packages/engine-server/src/workspace/service.ts
Comment (jonathanyeung) [inline packages/engine-server/src/workspace/service.ts:1596]: let's just delete if we don't need it?

**Review by jonathanyeung — COMMENTED** [review]
lgtm apart from some merge conflicts.  good to merge if CI pass looks clean, nice job!

File: packages/engine-server/src/DendronEngineV3.ts
Comment (kevinslin) [inline packages/engine-server/src/DendronEngineV3.ts:982]: can we add a next task to get rid of this option? caching should always be on

File: packages/engine-server/src/enginev2.ts
Comment (kevinslin) [inline packages/engine-server/src/enginev2.ts:105]: can we just remove this?

**Review by kevinslin — APPROVED** [review]
awesome job. just some small comments. feel free to merge after fixing the merge conflict

File: packages/engine-server/src/workspace/service.ts
Comment (hikchoi) [reply packages/engine-server/src/workspace/service.ts:1596]: I left it there to move / re-use it in the still-pending task to handle v4 configs gracefully, but yeah, this will be cleaned up.

File: packages/engine-server/src/DendronEngineV3.ts
Comment (hikchoi) [reply packages/engine-server/src/DendronEngineV3.ts:982]: Git it. I left it there to avoid scope creep.

**Comment by hikchoi** [comment]
Just a clarification that this is still incomplete and we probably should not merge this before we handle the v4 users.

I'll re-request a review after that part is implemented.

**Comment by kevinslin** [comment]
can we merge this and then handle v4 users as the next task? (as long as get it in by the next release in two weeks time, this should be good)

**Pull request merged**
